### PR TITLE
Add support for wildcards which match any property name.

### PR DESCRIPTION
**Pull request by mjball**
@HiJon89

### CODE REVIEW

File: src/main/java/com/hubspot/jackson/jaxrs/PropertyFilter.java [hunk 1]
```diff
@@ -41,9 +49,10 @@ public void addProperty(String property) {
         property = property.substring(1);
       }
 
-      if (property.contains(".")) {
-        String prefix = property.substring(0, property.indexOf('.'));
-        String suffix = property.substring(property.indexOf('.') + 1);
+      int indexOfDot = property.indexOf('.');
```
Comment (mjball) [inline src/main/java/com/hubspot/jackson/jaxrs/PropertyFilter.java]: This change is just a minor performance optimization.

File: src/test/java/com/hubspot/jackson/jaxrs/PropertyFilterTest.java [hunk 2]
```diff
@@ -60,14 +61,74 @@ public void testNestedExclude() {
...
+  @Test
+  public void testReadmeWildcardInclusion() throws Exception {
+    String json = "{" +
+        "  \"id\": 54," +
+        "  \"name\": \"Object\"," +
+        "  \"child\": {" +
+        "    \"id\": 96," +
+        "    \"name\": \"Child Object\"" +
+        "  }" +
+        "}";
```
Comment (mjball) [inline src/test/java/com/hubspot/jackson/jaxrs/PropertyFilterTest.java]: I find this style of creating test input JSON to be clearer & less verbose than programmatically creating `ObjectNode`s.

File: src/main/java/com/hubspot/jackson/jaxrs/PropertyFilter.java [hunk 3]
```diff
@@ -83,19 +92,38 @@ private void filter(ArrayNode array) {
...
 
-      object.remove(excludedProperties);
+      if (excludedProperties.contains(WILDCARD)) {
+        object.removeAll();
+      } else {
+        object.remove(excludedProperties);
+      }
 
       for (Entry<String, NestedPropertyFilter> entry : nestedProperties.entrySet()) {
-        JsonNode node = object.get(entry.getKey());
```
Comment (HiJon89) [inline src/main/java/com/hubspot/jackson/jaxrs/PropertyFilter.java]: `final`?

Comment (mjball) [inline src/main/java/com/hubspot/jackson/jaxrs/PropertyFilter.java]: Whoops, I added that when I was playing around with lookahead but tossed it all out.

File: src/main/java/com/hubspot/jackson/jaxrs/PropertyFilter.java [hunk 4]
```diff
@@ -83,19 +92,38 @@ private void filter(ArrayNode array) {
...
+        final Iterable<JsonNode> nodes;
```
Comment (mjball) [inline src/main/java/com/hubspot/jackson/jaxrs/PropertyFilter.java]: Done &uarr;

**Comment by mjball** [comment]
This feature is incompletely implemented in this branch and would likely take a significant rewrite to do correctly. Fortunately it turns out that I don't need this yet, so tabling for the forseeable future.

**Pull request closed without merging**